src/evaluate/hub.py

```diff
@@ -0,0 +1,125 @@
+from typing import Dict
+
+import requests
+from datasets.utils.metadata import known_task_ids
+from huggingface_hub import dataset_info, model_info
+from huggingface_hub.repocard import metadata_update
+
+from .utils.logging import get_logger
+
+
+logger = get_logger(__name__)
+
+
+def get_allowed_tasks(tasks_dict):
+    return list(tasks_dict.keys())
+
+
+def push_to_hub(
+    repo_id: str,
+    task_type: str,
+    dataset_type: str,
+    dataset_name: str,
+    metric_type: str,
+    metric_name: str,
+    metric_value: float,
+    task_name: str = None,
+    dataset_config: str = None,
+    dataset_split: str = None,
+    dataset_revision: str = None,
+    dataset_args: Dict[str, int] = None,
+    metric_config: str = None,
+    metric_args: Dict[str, int] = None,
+    overwrite: bool = False,
+):
+    r"""
+    Pushes the result of a metric to the Hub, adding it to the specified model's card.
+
+    Args:
+        repo_id (``str``): Model id from https://hf.co/datasets.
+        task_type (``str``): Task id, refer to
+            https://github.com/huggingface/datasets/blob/master/src/datasets/utils/resources/tasks.json for allowed values.
+        dataset_type (``str``): Dataset id from https://hf.co/datasets.
+        dataset_name (``str``): Pretty name for the dataset.
+        metric_type (``str``): Metric id from https://hf.co/metrics.
+        metric_name (``str``): Pretty name for the metric.
```

Comment:
@lvwerra is there anything in `evaluate` we can use to get the pretty name for the metric? This would make it even simpler to push metrics to the Hub.

Reply:
Unfortunately not, the pretty name is not part of the module at this stage.